client/internal/clientcommon.go

```diff
@@ -191,9 +192,27 @@ func (c *ClientCommon) SetAgentDescription(descr *protobufs.AgentDescription) er
 // calcHashEffectiveConfig calculates and sets the Hash field from the rest of the
 // fields in the message.
 func calcHashEffectiveConfig(msg *protobufs.EffectiveConfig) {
+	cfgMap := msg.GetConfigMap().GetConfigMap()
+
+	// If no cfgMap return empty hash
+	if cfgMap == nil || len(cfgMap) == 0 {
```

Comment:
The spec does not prohibit empty configurations. Empty map is a valid (empty) configuration. We need to calc the hash in that case.

Reply:
So the hash would have to just be `msg.Hash = h.Sum(nil)` since there's nothing to add to the hash. I avoided that since it isn't determined by the config at all. I wasn't sure if `h.Sum(nil)` is better than just empty byte array.

I have no problem switching it to an "empty" hash but I just wanted to throw out my thought process.

Reply:
The spec [requires](https://github.com/open-telemetry/opamp-spec/blob/main/specification.md#agent-to-server-state-synchronization) that the Hash field is always set:
>The compression is done by omitting all fields in the message, except the hash field which MUST always be present (see below for how the hash field is used).

`h.Sum(nil)` is a fine value to indicate an empty config.

Reply:
@tigrannajaryan updated. I still had it shortcut around to not cause a panic in the case of a nil configMap